src/Adapter/Discount/Update/DiscountConditionsUpdater.php

```diff
@@ -305,4 +311,58 @@ private function cleanDiscountCountries(CartRule $discount): array
 
         return ['country_restriction'];
     }
+
+    /**
+     * Update customer group restrictions for a discount.
+     *
+     * @param CartRule $discount
+     * @param int[] $customerGroupIds
+     *
+     * @return array List of updated properties
+     */
+    private function applyCustomerGroupConditions(CartRule $discount, array $customerGroupIds): array
+    {
+        $this->cleanCustomerGroups($discount);
```

Comment:
I think this one is redundant since it is already called in `cleanAllConditions`